### PR TITLE
Add test for per-partition stored-field format

### PR DESCRIPTION
**Pull request by romseygeek**
Tests for https://github.com/crate/crate/pull/17041

### CODE REVIEW

**Comment by romseygeek** [comment]
Note that the new test fails because we need to merge https://github.com/crate/crate/pull/17041 first

File: tests/bwc/test_partition_storage_formats.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    )
+    
+def _partition_paths(cursor):
+    paths = defaultdict(list)
+    cursor.execute('select p.values, s.path from sys.shards s, information_schema.table_partitions p where s.partition_ident = p.partition_ident;')
+    results = cursor.fetchall()
+    for (values, path) in results:
+        version = values['version']
+        paths[version].append(path)
+    pprint.pp(paths)
```
Comment (mfussenegger) [inline tests/bwc/test_partition_storage_formats.py:27]:
```suggestion
```

File: tests/bwc/test_partition_storage_formats.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    size = 0
+    if not data_path:
+        return 0
+    for dirpath, dirnames, filenames in os.walk(data_path):
+        for filename in filenames:
+            path = Path(dirpath, filename)
+            ext = path.suffix.lstrip('.')
+            if ext == 'fdt':
+                filesize = path.stat().st_size
+                print(f"{path}: {filesize}")
```
Comment (mfussenegger) [inline tests/bwc/test_partition_storage_formats.py:40]:
```suggestion
```

File: tests/bwc/test_partition_storage_formats.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                self._run_tests(UPGRADE_PATH, nodes=3)
+            finally:
+                self.tearDown()       
+        
+    def _run_tests(self, upgrade_path, nodes):
+        
+        # start initial cluster
+        settings = {
+            "transport.netty.worker_count": 16
+        }
```
Comment (mfussenegger) [inline tests/bwc/test_partition_storage_formats.py:59]: Afaik we set the worker count in the other bwc tests due to some bug in an older version. Shouldn't be necessary to set this here.

File: tests/bwc/test_partition_storage_formats.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        cluster = self._new_cluster(upgrade_path.from_version, nodes, settings=settings)
+        cluster.start()
+        with connect(cluster.node().http_url, error_trace=True) as conn:
+            c = conn.cursor()
+            # created partitioned table
```
Comment (mfussenegger) [inline tests/bwc/test_partition_storage_formats.py:64]:
```suggestion
```

File: tests/bwc/test_partition_storage_formats.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+            c.execute(f'''
+                CREATE TABLE partitioned (
+                    version STRING,
+                    value INTEGER
+                )
+                PARTITIONED BY (version)
+            ''')
+            # add data
```
Comment (mfussenegger) [inline tests/bwc/test_partition_storage_formats.py:72]:
```suggestion
```

File: tests/bwc/test_partition_storage_formats.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+            _add_data(c, [ upgrade_path.from_version ], 500)
+            
+        
+        # upgrade
```
Comment (mfussenegger) [inline tests/bwc/test_partition_storage_formats.py:76]:
```suggestion
```

File: tests/bwc/test_partition_storage_formats.py [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+            old_size = sum(_fdt_size(path) for path in partition_paths[upgrade_path.from_version])
+            new_size = sum(_fdt_size(path) for path in partition_paths[upgrade_path.to_version])
+            
+            print(f'Old partition has {old_size} bytes of FDT data')
+            print(f'New partition has {new_size} bytes of FDT data')
+            
+            self.assertLess(new_size, old_size * 0.25, f'Expected new partition FDT size {new_size} to be less than 25% of old partition {old_size}')
+        
+        
+
```
Comment (mfussenegger) [inline tests/bwc/test_partition_storage_formats.py:110]:
```suggestion
```

**Comment by romseygeek** [comment]
Somewhat confusingly, this PR has actually passed the tests here: https://jenkins.crate.io/job/CrateDB/job/qa/job/crate_qa_on_pr/436/

**Review by mfussenegger — APPROVED** [review]

**Comment by romseygeek** [comment]
retest this please

**Pull request merged**